docs/compaction-merge-loops.md

```diff
@@ -0,0 +1,153 @@
+# Compaction Merge Loops
+
```

Comment:
This does not include prefix tombstone propagation.  I will update for that after we agree on basic correctness of this.